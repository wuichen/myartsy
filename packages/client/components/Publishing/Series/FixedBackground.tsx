import React from "react"
import styled, { StyledFunction } from "styled-components"

interface Props {
  backgroundColor?: string
  backgroundUrl?: string
}

export const FixedBackground: React.SFC<Props> = props => {
  const { backgroundColor, backgroundUrl } = props

  if (backgroundUrl && !backgroundUrl.includes(".mp4")) {
    return (
      <FixedBackgroundContainer backgroundColor={backgroundColor}>
        <BackgroundImage backgroundUrl={backgroundUrl} />
        <BackgroundGradient />
      </FixedBackgroundContainer>
    )
  }
  return <FixedBackgroundContainer backgroundColor={backgroundColor} />
}

const Div: StyledFunction<Props & React.HTMLProps<HTMLDivElement>> = styled.div

export const FixedBackgroundContainer = Div`
  background-color: ${props => props.backgroundColor};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
`
const BackgroundImage = Div`
  background-image: ${props => `url(${props.backgroundUrl})`};
  background-size: cover;
  background-position: 50%;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -2;
`

const BackgroundGradient = Div`
  background: linear-gradient(rgba(0,0,0,0.3), rgba(0,0,0,0.6));
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
`
