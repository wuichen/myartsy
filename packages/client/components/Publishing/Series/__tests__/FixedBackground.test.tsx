import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { FixedBackground } from "../FixedBackground"

const imageURL = "https://foo.com/image.png"
const videoURL = "https://foo.com/video.mp4"

describe("FixedBackground", () => {
  it("renders the provided background color if no image available", () => {
    const component = renderer
      .create(<FixedBackground backgroundColor={"red"} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it("renders the provided background image", () => {
    const component = renderer
      .create(
        <FixedBackground backgroundColor={"red"} backgroundUrl={imageURL} />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it("ignores video for now", () => {
    const component = renderer
      .create(
        <FixedBackground backgroundColor={"red"} backgroundUrl={videoURL} />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
