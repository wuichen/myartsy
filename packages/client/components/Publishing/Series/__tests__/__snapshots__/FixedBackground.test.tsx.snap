// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedBackground ignores video for now 1`] = `
.c0 {
  background-color: red;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

<div
  className="c0"
/>
`;

exports[`FixedBackground renders the provided background color if no image available 1`] = `
.c0 {
  background-color: red;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

<div
  className="c0"
/>
`;

exports[`FixedBackground renders the provided background image 1`] = `
.c0 {
  background-color: red;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

.c1 {
  background-image: url(https://foo.com/image.png);
  background-size: cover;
  background-position: 50%;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -2;
}

.c2 {
  background: linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.6));
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
